src/test/java/BaseTest.java

```diff
@@ -1,11 +1,146 @@
 import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.chrome.ChromeOptions;
-import org.testng.annotations.BeforeSuite;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.FluentWait;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.*;
+
+import org.openqa.selenium.interactions.Actions;
+
+
+import java.time.Duration;
 
 public class BaseTest {
 
+    WebDriver driver;
+    WebDriverWait wait;
+    ChromeOptions options;
+    String url = null;// https://qa.koel.app/"; // "https://app.testpro.io/";
+    Actions action;
+    FluentWait fluentWait;
+
+    @DataProvider(name="LoginNegativeTestData")
+    public Object[][] getDataFromDataProvider(){
+        return new Object[][]{
+                {"invalidEmail@email.com","Invalid"},   // Invalid Email Data
+                {"",""},                               // Empty email and pwd
+        };
+    }
+
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
+
+    @BeforeMethod
+    @Parameters({"baseUrl"})
+    public void initMethod(String baseUrl){
+        options = new ChromeOptions();
+        options.addArguments("--remote-allow-origins=*");
```

Comment:
Add multiple arguments:
`--disable-notifications`: disabling notifications
`--remote-allow-origins=*`: allowing remote origins
`--incognito`: launching in incognito mode
`--start-maximized`: launching in maximized mode
`"excludeSwitches", new String[]{"enable-automation"} `: Disable notification Chrome is being controlled by automated test software in ChromeDriver
**These settings are enabled to simplify and improve the process of automated testing.**
_For example_
![image](https://github.com/Test-Pro-Bootcamp/A57-QA-Automation-Framework/assets/104761482/b05c1900-365a-491c-a399-92b5b3bdc666)
```
                WebDriverManager.chromedriver().setup();
                ChromeOptions options = new ChromeOptions();
                options.addArguments("--disable-notifications","--remote-allow-origins=*", "--incognito","--start-maximized");
                options.setExperimentalOption("excludeSwitches", new String[]{"enable-automation"});
                return new ChromeDriver(options);

```